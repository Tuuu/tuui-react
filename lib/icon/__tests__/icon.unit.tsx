import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Icon from '../icon';

describe('icon tests', () => {
  it('svg render', function() {
    const iconJson = renderer.create(<Icon name="apple" />).toJSON();
    expect(iconJson).toMatchSnapshot();
  });
  it('svg onClick', function() {
    const clickCallback = jest.fn(); //声明一个函数
    const click = mount(<Icon name="android" onClick={clickCallback} />);
    click.find('svg').simulate('click');
    expect(clickCallback).toBeCalled(); // 期待函数调用
  });
});
