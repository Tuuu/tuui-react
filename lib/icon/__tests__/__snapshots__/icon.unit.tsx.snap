// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon tests svg render 1`] = `
<svg
  className="tui-icon"
>
  <use
    xlinkHref="#apple"
  />
</svg>
`;
